cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/RunRecordMonitorService.java

```diff
@@ -166,11 +174,23 @@ private void cleanupQueue() {
       // Queue head might have already been removed. So instead of calling poll, we call remove.
       if (launchingQueue.remove(programRunId)) {
         LOG.info("Removing request with runId {} due to expired retention time.", programRunId);
+        emitMetrics(Constants.Metrics.FlowControl.FLOW_CONTROL_LAUNCHING_RUNS, launchingQueue.size());
```

Comment:
I think outstanding will also go down

Reply:
Thanks for catching this. Decided to change outstanding to only running metrics for simplicity.